src/services/codefixes/inferFromUsage.ts

```diff
@@ -201,15 +215,70 @@ namespace ts.codefix {
     }
 
     function annotateJSDocParameters(changes: textChanges.ChangeTracker, sourceFile: SourceFile, parameterInferences: ParameterInference[], program: Program, host: LanguageServiceHost): void {
-        const result = mapDefined(parameterInferences, inference => {
+        const signature = parameterInferences.length && parameterInferences[0].declaration.parent;
+        if (!signature) {
+            return;
+        }
+        const paramTags = createUnderscoreEscapedMap<JSDocParameterTag>();
+        forEach(parameterInferences, inference => {

```

Comment:
This is typed as a non-nullable array so this could just be a for loop? Also, this function never writes to `parameterInferences` so that could be a `ReadonlyArray`.